src/lib.rs

```diff
@@ -426,6 +426,142 @@ fn graph_is_isomorphic(
     Ok(res)
 }
 
+/// Determine if 2 directed graphs are subgraph - isomorphic
+///
+/// This checks if 2 graphs are subgraph isomorphic both structurally and also
+/// comparing the node data and edge data using the provided matcher functions.
+/// The matcher function takes in 2 data objects and will compare them. A simple
+/// example that checks if they're just equal would be::
+///
+///     graph_a = retworkx.PyDiGraph()
+///     graph_b = retworkx.PyDiGraph()
+///     retworkx.is_subgraph_isomorphic(graph_a, graph_b,
+///                                     lambda x, y: x == y)
+///
+/// .. note::
+///
+///     For better performance on large graphs, consider setting `id_order=False`.
+///
+/// :param PyDiGraph first: The first graph to compare
+/// :param PyDiGraph second: The second graph to compare
+/// :param callable node_matcher: A python callable object that takes 2 positional
+///     one for each node data object. If the return of this
+///     function evaluates to True then the nodes passed to it are vieded
+///     as matching.
+/// :param callable edge_matcher: A python callable object that takes 2 positional
+///     one for each edge data object. If the return of this
+///     function evaluates to True then the edges passed to it are vieded
+///     as matching.
+/// :param bool id_order: If set to ``False`` this function will use a
+///     heuristic matching order based on [VF2]_ paper. Otherwise it will
+///     default to matching the nodes in order specified by their ids.
+///
+/// :returns: ``True`` if there is a subgraph of `first` isomorphic to `second`,
+///     ``False`` if there is not.
+/// :rtype: bool
+#[pyfunction(id_order = "true")]
```

Comment:
Since this is a new function we do have the flexability to change the default here and enable the vf2++ heuristic by default. I'll defer to what you think is best, either consistency with `is_isomorphic` or having the generally faster heuristic used by default.